### ISSUE TITLE
Missing compiler

### ISSUE THREAD

**Issue by giavani1947** [tessel/t2-cli#1644]
I've got the following issue:

WARN Pre-compiled module is missing: @serialport/bindings@2.0.8.
WARN Please file an issue at https://github.com/tessel/t2-cli/issues/new with th
is warning.
WARN
WARN This warning might be caused by one of the following:
WARN
WARN 1. A pre-compiled binary has not yet been built for this module.
WARN 2. The binary didn't compile correctly for the platform that you're develop
ing on.
WARN    Binaries that are Linux-only or even OpenWRT-specific may cause this iss
ue.
WARN    Try to npm install --force the affected module, then rerun your deployme
nt command.
WARN 3. The binary may be platform specific and impossible to compile for OpenWR

Any help thanks.

**Comment by HipsterBrown** [comment]
@giavani1947 What version of Node.js are you using on Tessel? Can you share the output from `t2 version` while you have your Tessel connected?

**Comment by giavani1947** [comment]
My output for the t2 version:
Your environment has been set up for using Node.js 10.15.3 (x64) and npm.

C:\Users\Trong Huynh>t2 version
INFO Looking for your Tessel...
INFO Connected to t3.
INFO Tessel Environment Versions:
INFO t2-cli: 0.1.22
INFO t2-firmware: 0.2.0
INFO Node.js: 8.11.3

Also, below is the output I get from running the t2 led.js program before it stalls without the led blinking:

C:\Users\Trong Huynh>t2 run led.js
INFO Looking for your Tessel...
INFO Connected to t3.
WARN ENOENT: no such file or directory, lstat 'led.js'

C:\Users\Trong Huynh>cd\

C:\>cd\users/trong huynh\j5ik

C:\Users\Trong Huynh\j5ik>t2 run led.js
INFO Looking for your Tessel...
INFO Connected to t3.
INFO Building project.
WARN Pre-compiled module is missing: @serialport/bindings@2.0.8.
WARN Please file an issue at https://github.com/tessel/t2-cli/issues/new with th
is warning.
WARN
WARN This warning might be caused by one of the following:
WARN
WARN 1. A pre-compiled binary has not yet been built for this module.
WARN 2. The binary didn't compile correctly for the platform that you're develop
ing on.
WARN    Binaries that are Linux-only or even OpenWRT-specific may cause this iss
ue.
WARN    Try to npm install --force the affected module, then rerun your deployme
nt command.
WARN 3. The binary may be platform specific and impossible to compile for OpenWR
T.
INFO Writing project to RAM on t3 (547.84 kB)...
INFO Deployed.
INFO Running led.js...
1565884695388 Available Tessel 2 (t3)
1565884695518 Connected Tessel 2 (t3)
1565884695567 Repl Initialized
>>

**Comment by HipsterBrown** [comment]
Alright I'll test this out at some point this week.

**Comment by HipsterBrown** [comment]
@giavani1947 I tested this out today while running Node v10.15.3 and Node v8.11.3 locally, and both times I was able to use `johnny-five`, which requires `@serialport/bindings@v2.0.8`. Can you check your home directory for a `.tessel` folder (or directory) with a `binaries` folder inside that? In that binaries folder, there should be a `@serialport` folder with a file that has `bindings-2.0.8` in the name. 

We definitely have that pre-compiled binary available. Can you also make sure you're connected to the internet when running the program, in case the CLI needs to fetch that binary?

**Comment by giavani1947** [comment]
Yes, the bindings folder is there but it is titled:
bindings-2.0.8-Release-node-v57-linux-mipsel
I am running this on windows not linus so would this be the issue?

Also, does this need to be in the same directory as the experiment programs being ran?

**Comment by HipsterBrown** [comment]
> Also, does this need to be in the same directory as the experiment programs being ran?

No, the CLI should automatically look in that `.tessel/binaries` directory in your home directory for any matching binaries required by your program. 

It _could_ be an issue with looking up that binary on Windows, I'll look into that.

**Comment by alex-engelmann** [comment]
I'm having this exact same issue, in Windows 10 x64.  Exact same version numbers on all the above, same errors, same directory contents.  I was about to make a new issue like the instructions said, but I found this thread.  I got the led.js program loaded and running fine despite these warnings and a few other errors.
